File: android/app/src/main/kotlin/com/runtrack/data/models/Shoe.kt
package com.runtrack.data.models

import kotlinx.serialization.Serializable

@Serializable
data class Shoe(
    // TODO: PM을 참고하여 신발 모델 단연 정의
)

@Serializable
data class ShoeCreateRequest(
    // TODO: 신발 등록 숲고 단연 정의
)
    val notes: String? = null,
)

@Serializable
data class ShoeUpdateRequest(
    val brand: String? = null,
    val model: String? = null,
    val size: String? = null,
    val color: String? = null,
    val purchasePriceWon: Int? = null,
    val condition: String? = null,
    val notes: String? = null,
)

@Serializable
data class ShoeStats(
    val cumulativeKm: Float,
    val runCount: Int,
    val condition: String,
)

File: backend/app/schemas/shoe.py
"""
신발 Pydantic 스키마
"""
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class ShoeBase(BaseModel):
    brand: str
    model: str
    size: Optional[str] = None
    color: Optional[str] = None
    purchase_price_won: Optional[int] = None
    purchase_date: Optional[datetime] = None
    notes: Optional[str] = None

class ShoeCreate(ShoeBase):
    pass

class ShoeUpdate(BaseModel):
    brand: Optional[str] = None
    model: Optional[str] = None
    size: Optional[str] = None
    color: Optional[str] = None
    purchase_price_won: Optional[int] = None
    condition: Optional[str] = None
    notes: Optional[str] = None

class ShoeStats(BaseModel):
    cumulative_km: float
    run_count: int
    condition: str

class ShoeResponse(ShoeBase):
    id: int
    user_id: int
    cumulative_km: float
    run_count: int
    condition: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
